packages/pg/lib/client.js

```diff
@@ -198,6 +204,38 @@ class Client extends EventEmitter {
     con.on('notification', this._handleNotification.bind(this))
   }
 
+  async _handleGSSInit(msg) {
+    try {
+      this.client = await kerberos.initializeClient(`${this.principal}@${this.host}`, {
+        mechOID: kerberos.GSS_MECH_OID_SPNEGO,
+      })
+
+      // TODO: below this might need to be a recursive loop to step multiple times.
+      const token = await this.client.step('')
+
+      const buf = Buffer.from(token, 'base64')
+      this.connection.sendBinaryPassword(buf)
+    } catch (e) {
+      this.emit('error', e)
+    }
+  }
+
+  async _handleGSSContinue(msg) {
+    try {
+      const inToken = msg.inToken
+      const token = await this.client.step(inToken)
+
+      // TODO: probably a better way to handle this.
+      if (token == null) {
+        this.emit('error', 'Received null GSSAPI token on continue')
```

Comment:
Just a quick note; the token can be null if the negotiation has already been successfully completed. I suspect the right thing to do is to do something like:

```
if (inToken === 'oRQwEqADCgEAoQsGCSqGSIb3EgECAg==') {
  return; // negotiation successful, no need to do anything further.
}

see: https://github.com/jcmturner/gokrb5/blob/master/spnego/http.go#L198 for where that magic string came from.